src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -638,6 +638,7 @@ void RegisterAppInterfaceRequest::SendRegisterAppInterfaceResponseToMobile() {
   SendOnAppRegisteredNotificationToHMI(
       *(application.get()), resumption, need_restore_vr);
 
+  GetPolicyHandler().AddApplication(application->policy_app_id());
```

Comment:
@LuxoftAKutsan 
By doing that (reverting PR #1137 ) you'll get APPLINK-30884 reopened again since order of OnAppRegistered and OnStatusUpdate will be changed.